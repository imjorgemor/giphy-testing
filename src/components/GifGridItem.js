import PropTypes from 'prop-types';

const GifGridItem = ({ url, title}) => {

    return (
        <div className='card'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}