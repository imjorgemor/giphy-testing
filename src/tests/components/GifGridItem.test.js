import { shallow } from "enzyme"
import GifGridItem from "../../components/GifGridItem"


describe("prueba en gifgrid Item", () => {

    const title = "un titulo";
    const url = "https://localhost.3333"
    const component = shallow(<GifGridItem title={title} url={url} />)

    test("debe mostar el componente, test de render", () => {
        expect(component).toMatchSnapshot();
    })

    test("debe de tener un pÃ¡rrafo con el title", () => {
        const p = component.find('p')
        expect(p.text().trim()).toBe(title)
    })

    test('debe de tener la imagen igual al url y alt', () => {

        const img = component.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop("alt")).toBe(title);
    })

    test("debe tener class card", ()=> {
        // analizar si tiene una clase concreta
        const div = component.find('div');
        const className = div.prop('className');
        expect(className.includes("card")).toBe(true);
    })
})