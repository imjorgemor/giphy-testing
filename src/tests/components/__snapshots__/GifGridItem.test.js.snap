// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en gifgrid Item debe mostar el componente, test de render 1`] = `
<div
  className="card"
>
  <img
    alt="un titulo"
    src="https://localhost.3333"
  />
  <p>
    un titulo
  </p>
</div>
`;
